docs/tutorials/provider/README.md

```diff
@@ -0,0 +1,125 @@
+
+1. A digital twin consumer should utilize the `Subscribe` operation to consume the *ambient air temperature* and the *is air conditioning active* in-vehicle signals.
+
+1. A digital twin consumer should utilize the `Set` operation to set the value of an in-vehicle signal.
+
+1. A digital twin consumer should utilize the `Invoke` operation to send a *show notification* command.
+
+### 1.2 Implement the Operations for the Digital Twin Provider Interface
+
+You have defined your [digital twin provider interface](../../../samples/interfaces/sample_grpc/v1/digital_twin_provider.proto). Let's implement the functionality for the `Subscribe`, `Set` and `Invoke` operations.
+
+#### Implement the Operations for the Sample Digital Twin Provider Interface
+
+1. Reference the [code for implementing the operations for the sample digital twin provider interface](../../../samples/mixed/provider/src/provider_impl.rs). Please only consider the implementations for the `Subscribe`, `Set` and `Invoke` operations.
+
+1. There is an import statement for the in-vehicle digital model that you have previously constructed in the [Tutorial: Create an In-Vehicle Model with DTDL](../in_vehicle_model/README.md):
+
+```rust
+use digital_twin_model::sdv_v1 as sdv;
+```
+
+1. The implementation of the `Set` operation references the signals *is air conditioning active* and *ambient air temperature*.
+
+1. The implementation of the `Invoke` operation references the command *show notification*.
+
+## 2. Register Digital Twin Provider with the In-Vehicle Digital Twin Service
+
+You have defined a sample interface with the following operations: `Subscribe`, `Set` and `Invoke` operations. You have implemented the functionality for each operation.
+
+You will need to register your digital twin provider with the [In-Vehicle Digital Twin Service](../../../README.md#high-level-design). This registration will make your digital twin provider discoverable to digital twin consumers through the In-Vehicle Digital Twin Service.
+
+>[In-Vehicle Digital Twin Service:](../../../README.md#high-level-design) "Ibeji's architecture has an In-Vehicle Digital Twin Service at its core. The In-Vehicle Digital Twin Service captures all of the vehicle's primary capabilities and makes them available to Ibeji consumers".
+
+1. Reference the [main.rs file of your digital twin provider](../../../samples/mixed/provider/src/main.rs). The main.rs file outlines the behavior of the signals in your digital twin provider sample. This includes a vehicle simulator that can emulate changes in its signals. These changes are then published to any digital twin consumers that have subscribed to your digital twin provider.
+
+1. One function of particular interest in the [main.rs](../../../samples/mixed/provider/src/main.rs) file is the register function.
+
+```rust
```

Comment:
Is it wise to include Rust helper functions in this tutorial?  Shouldn't this tutorial also help those using languages other than Rust?  I would focus more on the gRPC proto files.

Reply:
I've added some Rust example sections, and explicitly highlighted that other programming languages can be used.

Reply:
If someone makes a sample in a different language we could always ask them to provide an example in that language here, I would say having some example code is better than no code.

Reply:
I added your suggestion @devkelley! I've included new sample tutorial code for a provider and consumer.